app/src/main/java/com/nextcloud/talk/utils/BitmapShrinker.kt

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * https://developer.android.com/topic/performance/graphics/load-bitmap.
+ * https://stackoverflow.com/a/15341203
+ */
+object BitmapShrinker {
+
+    private val TAG = "BitmapShrinker"
+
+    fun shrinkBitmap(
+        path: String,
+        reqWidth: Int,
+        reqHeight: Int
+    ): Bitmap {
+        val bitmap = decodeSampledBitmap(path, reqWidth, reqHeight)
+        return rotateBitmap(path, bitmap)
+    }
+
+    private fun decodeSampledBitmap(
+        path: String,
+        reqWidth: Int,
+        reqHeight: Int
+    ): Bitmap {
+        // First decode with inJustDecodeBounds=true to check dimensions
+        return BitmapFactory.Options().run {
+            inJustDecodeBounds = true
+            // BitmapFactory.decodeResource(res, resId, this)
+            BitmapFactory.decodeFile(path, this)
+
+            // Calculate inSampleSize
+            inSampleSize = calculateInSampleSize(this, reqWidth, reqHeight)
+
+            // Decode bitmap with inSampleSize set
+            inJustDecodeBounds = false
+
+            // BitmapFactory.decodeResource(res, resId, this)
+            BitmapFactory.decodeFile(path, this)
+        }
+    }
+
+    private fun calculateInSampleSize(options: BitmapFactory.Options, reqWidth: Int, reqHeight: Int): Int {
+        // Raw height and width of image
+        val (height: Int, width: Int) = options.run { outHeight to outWidth }
+        var inSampleSize = 1
+
+        if (height > reqHeight || width > reqWidth) {
+
+            val halfHeight: Int = height / 2
+            val halfWidth: Int = width / 2
+
+            // Calculate the largest inSampleSize value that is a power of 2 and keeps both
+            // height and width larger than the requested height and width.
+            while (halfHeight / inSampleSize >= reqHeight || halfWidth / inSampleSize >= reqWidth) {
+                inSampleSize *= 2
+            }
+        }
+        return inSampleSize
+    }
+
+    private fun rotateBitmap(path: String, bitmap: Bitmap): Bitmap {
+        try {
+            val exif = ExifInterface(path)
+            val orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, 1)
+            val matrix = Matrix()
+            when (orientation) {
+                ExifInterface.ORIENTATION_ROTATE_90 -> {
+                    matrix.postRotate(90f)
+                }
+                ExifInterface.ORIENTATION_ROTATE_180 -> {
+                    matrix.postRotate(180f)
+                }
+                ExifInterface.ORIENTATION_ROTATE_270 -> {
+                    matrix.postRotate(270f)
+                }
+            }
+            val rotatedBitmap = Bitmap.createBitmap(
+                bitmap,
+                0,
+                0,
+                bitmap.getWidth(),
+                bitmap.getHeight(),
+                matrix,
+                true
+            )
+            return rotatedBitmap
+        } catch (e: IOException) {
+            Log.e(TAG, "error while rotating image")
```

Comment:
I'd add the exception as a 3rd parameter

Reply:
> Couldn't reproduce an issue before, but code is in the files app, so should be fine.

infact i went for an other solution: https://developer.android.com/topic/performance/graphics/load-bitmap